File: removeonly_firstegg.py
def remove_egg (foods):
    result = []
    egg_count = 0 # counter for eggs removed

    for i in range(len(foods)):
        if foods [i] == 'egg' and egg_count<2:
            egg_count +=1
            continue
        result.append (foods[i])
    return result
print(remove_egg(['egg', 'apple', 'egg', 'egg', 'ham']))




    

File: removeonly_firstegg.js
function removeEgg(foods) {
    const result = [];
    let eggCount = 0; // Counter for eggs removed
  
    for (let i = 0; i < foods.length; i++) {
      if (foods[i] === 'egg' && eggCount < 2) {
        eggCount++; // Increment counter and skip this 'egg'
        continue;
      }
      result.push(foods[i]); // Add non-'egg' items or excess 'egg' items
    }
    return result;
  }
  
  // Example:
console.log(removeEgg(['egg', 'apple', 'egg', 'egg', 'ham'])); // Output: ['apple', 'egg', 'ham']
  